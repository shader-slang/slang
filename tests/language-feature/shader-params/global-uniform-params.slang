// global-uniform-params.slang

//DISABLED_TEST(compute):COMPARE_COMPUTE:

// Test that code can use uniform parameters
// of "ordinary" type declared at the global scope

//TEST_INPUT:cbuffer(data=[256 1]):name=$Globals
uniform int a;
uniform int b;

int test(int val)
{
	return a*(val+1) + b*(val+2);
}

//TEST_INPUT:ubuffer(data=[0 0 0 0], stride=4):out,name=outputBuffer
RWStructuredBuffer<int> outputBuffer;

[numthreads(4, 1, 1)]
[shader("compute")]
void computeMain(uint3 dispatchThreadID : SV_DispatchThreadID)
{
    uint tid = dispatchThreadID.x;
    int inVal = tid;
    int outVal = test(inVal);
    outputBuffer[tid] = outVal;
}
