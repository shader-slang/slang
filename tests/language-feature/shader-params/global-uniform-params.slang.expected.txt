102
203
304
405
